### PR TITLE
Last Processing Job Duration shows most significant time units first

### PR DESCRIPTION
**Pull request by AMZN-Olex**
Signed-off-by: Olex Lozitskiy <5432499+AMZN-Olex@users.noreply.github.com>

## What does this PR do?

Changes Asset Processor job time field of "Last Processing Job Duration" to most significant time units first (ISO-8601 style).
image

Without this change, the time units are shown in least significant order.
image

## How was this PR tested?

Ran AP locally.

### CODE REVIEW

**Comment by AMZN-stankowi** [comment]
I'd like to get UX eyes on this. The current display was specifically chosen by UX after some experimenting.

**Review by lumberyard-employee-dm — APPROVED** [review]

**Comment by amzn-leenguy** [comment]
Tagging @rainbj

**Review by rainbj — APPROVED** [review]
UX approved. Changes look good. Thanks

**Review by cgalvan — APPROVED** [review]

**Review by AMZN-stankowi — APPROVED** [review]

**Comment by AMZN-Olex** [comment]
FYI, I went with `hh:mm:ss.zzz` format to fit the rest of the use of AP. Here is a screenshot for context.
image

File: Code/Tools/AssetProcessor/native/resourcecompiler/JobsModel.cpp [hunk 1]
```diff
@@ -195,19 +195,8 @@ namespace AssetProcessor
...
-                    {
-                        return processDuration.toString("zzz' ms, 'ss' sec, 'mm' min'");
-                    }
-                    if (processDuration.second() > 0)
-                    {
-                        return processDuration.toString("zzz' ms, 'ss' sec'");
-                    }
-                    return processDuration.toString("zzz' ms'");
+
+                    return processDuration.toString("hh:mm:ss.zzz");
```
Comment (lumberyard-employee-dm) [inline Code/Tools/AssetProcessor/native/resourcecompiler/JobsModel.cpp:199]: FYI I added a [GetUTCTimestampMilliseconds](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/std/time.h#L94-L97) function to `<AzCore/std/time.h>`

We may want to add a function there for hours minutes and seconds which is just half of the `GetUTCTimestamp*` function implementation

Comment (AMZN-Olex) [reply Code/Tools/AssetProcessor/native/resourcecompiler/JobsModel.cpp:199]: Yeah, we need a duration API not a whole timestamp here.

Comment (lumberyard-employee-dm) [reply Code/Tools/AssetProcessor/native/resourcecompiler/JobsModel.cpp:199]: Well it looks like QTime covers this so no need to do anything.
I was just providing a heads up if we needed to output a timestamp without using QT

**Review by AMZN-stankowi — APPROVED** [review]

**Pull request merged**